microservices/nris_api/backend/migrations/env.py

```diff
@@ -70,17 +71,26 @@ def process_revision_directives(context, revision, directives):
                                 poolclass=pool.NullPool)
 
     connection = engine.connect()
+    connection.dialect.default_schema_name = schema_name
     context.configure(connection=connection,
                       target_metadata=target_metadata,
                       process_revision_directives=process_revision_directives,
-                      **current_app.extensions['migrate'].configure_args)
+                      include_object=include_object,
+                      version_table_schema=schema_name,
+                      ** current_app.extensions['migrate'].configure_args)
 
     try:
         with context.begin_transaction():
             context.run_migrations()
     finally:
         connection.close()
 
+
+def include_object(object, name, type_, reflected, compare_to):
+    # Prevent alembic suicide
+    return name != "alembic_version"
```

Comment:
What's this for 😂

Reply:
alembic_version is the versioning table alembic uses to track which migrations have been executed. However, it tries to delete it next migration because there is no matching model.

Reply:
We were working around being stuck with using the same schema name as the username with SqlAlchemy. Alembic and SqlAlchemy weren't playing well with each other when we just used the 'table_args' parameter to specify the schema in the BaseModel object (Alembic was recreating FKs on every migrate). 

This line was part of code on the Alembic side to work with explicit and/or multiple schemas specified in a config file. 

We weren't able to finish it off from the SqlAlchemy side however. It might be worth keeping this around if we want to deal with multiple schemas, but I think we'll have to think through it a little further.